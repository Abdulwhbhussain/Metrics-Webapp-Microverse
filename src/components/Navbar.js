import React, { useState } from 'react';
import {
  NavLink, useNavigate, Link, useLocation,
} from 'react-router-dom';
import { TiWeatherPartlySunny } from 'react-icons/ti';
import '../styles/Navbar.css';
import { IoMdArrowRoundBack } from 'react-icons/io';
import { AiOutlineMenu } from 'react-icons/ai';
import { RxCross1 } from 'react-icons/rx';

const Navbar = () => {
  const location = useLocation();
  const detailPage = location.pathname.includes('/details/');
  const searchPage = location.pathname.includes('/search/');
  const [toggle, setToggle] = useState(false);
  const navigate = useNavigate();

  const handleClick = (e) => {
    e.preventDefault();
    navigate(-1);
  };

  return (
    <nav className={`flex md:flex ${detailPage || searchPage ? '' : 'flex-row'} md:flex-row justify-between px-2 md:px-10 py-5 items-center border-b-2 bg-sky-600`}>
      <Link to="/" className={`logo-container ${detailPage || searchPage ? 'hidden' : 'flex'}`}>
        <TiWeatherPartlySunny className="text-7xl text-white" />
      </Link>
      {detailPage || searchPage ? (
        <>
          <IoMdArrowRoundBack onClick={handleClick} className="text-4xl font-semibold text-white ms-10 border cursor-pointer border-white" />
          <div className="menu-container md:hidden block">
            <AiOutlineMenu className="text-white text-6xl pe-4" onClick={() => { setToggle(!toggle); }} />
          </div>
        </>
      ) : (
        <div className="menu-container md:hidden block">
          <AiOutlineMenu className="text-white text-6xl pe-4" onClick={() => { setToggle(!toggle); }} />
        </div>
      )}

      <ul className="md:flex  hidden">
        <li className="me-3"><NavLink to="/">Home</NavLink></li>
        <li className="me-3">
          <a href="https://openweathermap.org/api" target="_blank" rel="noopener noreferrer">
            Weather API
          </a>
        </li>
        <li className="me-3">
          <a href="https://openweathermap.org/api/geocoding-api" target="_blank" rel="noopener noreferrer">
            City API
          </a>
        </li>
      </ul>
      <div className={`flex mobile-list-container absolute slide-in ${toggle ? 'left-0 top-0' : 'left-[-100%] top-0'} bg-sky-300 z-20 `}>
        <ul className="m-10">
          <li className="my-5 text-4xl"><NavLink to="/" onClick={() => setToggle(!toggle)}>Home</NavLink></li>
          <li className="my-5 text-4xl">
            <a href="https://openweathermap.org/api" target="_blank" rel="noopener noreferrer">
              Weather API
            </a>
          </li>
          <li className="my-5 text-4xl">
            <a href="https://openweathermap.org/api/geocoding-api" target="_blank" rel="noopener noreferrer">
              City API
            </a>
          </li>
        </ul>
        <RxCross1 className="text-4xl text-white my-7" onClick={() => setToggle(!toggle)} />
      </div>

    </nav>
  );
};

export default Navbar;
