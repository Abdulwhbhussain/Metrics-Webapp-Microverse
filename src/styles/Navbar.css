nav {
  background-color: rgba(92, 92, 92, 0.108);
}

.active {
  color: white;
  border-bottom: 1px solid whitesmoke;
}

nav ul li {
  font-family: 'Roboto Slab', serif;
}

/* Add CSS for slide-in and slide-out transitions */
.mobile-list-container {
  height: 100vh;
  width: 100%;
}

.slide-in {
  transition: all 0.5s ease;
}

.header {
  background-color: rgba(92, 92, 92, 0.108);
  height: 100px;
  width: 100%;
  display: flex;
  justify-content: space-around;
  align-items: center;
  list-style-type: none;
  margin: 0;
  padding: 0;
  overflow: hidden;
  text-decoration: none;
}
