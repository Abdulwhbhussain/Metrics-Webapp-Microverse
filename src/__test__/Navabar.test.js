import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Navbar from '../components/Navbar';

describe('Navbar component', () => {
  test('renders correctly', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
