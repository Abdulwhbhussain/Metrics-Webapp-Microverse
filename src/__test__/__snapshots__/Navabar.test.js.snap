// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component renders correctly 1`] = `
<nav
  className="flex md:flex flex-row md:flex-row justify-between px-2 md:px-10 py-5 items-center border-b-2 bg-sky-600"
>
  <ul
    className="m-10 header"
  >
    <li
      className="my-5 text-4xl"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="my-5 text-4xl"
    >
      <a
        href="https://openweathermap.org/api"
        rel="noopener noreferrer"
        target="_blank"
      >
        Weather API
      </a>
    </li>
    <li
      className="my-5 text-4xl"
    >
      <a
        href="https://openweathermap.org/api/geocoding-api"
        rel="noopener noreferrer"
        target="_blank"
      >
        City API
      </a>
    </li>
  </ul>
</nav>
`;
